### ISSUE TITLE
Add typescript support

### ISSUE THREAD

**Issue by jenkoian** [jenkoian/hacktoberfest-checker#567]
Typescript is something I have very little experience of but like very much. We should implement it here if we can.

**Comment by vincentrohde** [comment]
Do you want to implement it for both api and client(`./src`) or only client?

**Comment by jenkoian** [comment]
@vincentrohde both? But maybe we do one at a time?

**Comment by vincentrohde** [comment]
I think one at a time would work good. I have a smaller Webpack project, where I implemented Typescript later on during development, just like in this case. Most of the Typescript I wrote so far was related to React/ front end code. If we would start with the client first, it would give me an easy start to get into the project and it's structure. Once that is done, the api could then be converted to Typescript. What do you think?

**Comment by EugeneGohh** [comment]
Hi, can I work on migrating the codebase to TypeScript too?

**Comment by vincentrohde** [comment]
@jenkoian I would love to do the backend, if we split it into two people doing the task separately. It would be something new for me. I would love to get a better understanding in typing an API 👍

**Comment by jenkoian** [comment]
Happy either way, are you ok to work it out amongst yourselves? Let me know if you want me to enable discussions on this repo or anything 👍

**Comment by vincentrohde** [comment]
@jenkoian I haven't used discussions before, but I guess it would be good to allow easier communication. @EugeneGohh we could also use Discord if you don't mind.

**Comment by jenkoian** [comment]
I actually think discussion is already enabled, but yeah whatever really. Also feel free to get a PR up, as often find that is good way to spark a discussion anyway 👍

**Comment by E-wave112** [comment]
Hello @jenkoian I'm also interested in working on the backend with typescript

**Comment by babu1998** [comment]
Try to avoid using any in typescript it would be better and more effective if the type was more specific
